org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/DiagnosticsHandler.java

```diff
@@ -134,11 +136,57 @@ public boolean isActive() {
 			diag.setCode(Integer.toString(problem.getID()));
 			diag.setSeverity(convertSeverity(problem));
 			diag.setRange(convertRange(openable, problem));
+			diag.setTags(getDiagnosticTag(problem.getID()));

```

Comment:
this should be set only if client supports it. See textDocument.publishDiagnostics.tagSupport. That requires lsp4j to be updated to read that information from the client capabilities

Reply:
Good catch. The whole thing is already working without setting that flag. I'll send PR in  LSP4J.

Reply:
Hmm, the PublishDiagnosticsClientCapabilities wad added 3 days ago. Later than the tag def. It did not exist when I implemented it.